streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -80,22 +80,30 @@ public void shouldLockTaskStateDirectory() throws Exception {
 
         directory.lock(taskId, 0);
 
-        final FileChannel channel = FileChannel.open(new File(taskDirectory, StateDirectory.LOCK_FILE_NAME).toPath(), StandardOpenOption.CREATE, StandardOpenOption.WRITE);
-        try {
+        try (
+            final FileChannel channel = FileChannel.open(
+                new File(taskDirectory, StateDirectory.LOCK_FILE_NAME).toPath(),
+                StandardOpenOption.CREATE, StandardOpenOption.WRITE)
+        ) {
             channel.tryLock();
             fail("shouldn't be able to lock already locked directory");
-        } catch (OverlappingFileLockException e) {
+        } catch (final OverlappingFileLockException e) {
             // pass
+        } finally {
+            directory.unlock(taskId);
```

Comment:
Rather than having `try{...} finally` everywhere it might be better to keep track of the locked `TaskId` and unlock them in the `cleanup` method.